crates/matrix-sdk-crypto/src/machine/tests/mod.rs

```diff
@@ -727,6 +732,174 @@ async fn test_megolm_encryption() {
     }
 }
 
+#[cfg(feature = "experimental-encrypted-state-events")]
```

Comment:
It would be nice to add a doc-comment explaining what this does. Something like: "Helper for testing Megolm encryption and decryption between two devices. Creates two devices, Alice and Bob, and has Alice create an outgoing Megolm session in the given room, whose decryption key is shared with Bob via a to-device message. Returns the two devices."

Reply:
Also: I think this is common to `test_megolm_encryption` as well? I don't think there is any problem with removing the `cfg` guard and using it from there: it will help break up that rather big test method into more digestible chunks.

Reply:
> Also: I think this is common to `test_megolm_encryption` as well? I don't think there is any problem with removing the `cfg` guard and using it from there: it will help break up that rather big test method into more digestible chunks.

`test_megolm_encryption` does some assertions on the `group_session` and `room_keys_received_stream` (which I removed from the state event tests in an earlier review pass), so I'm not sure it makes sense to refactor it as well?

Reply:
ok, fair enough :+1: